### PR TITLE
js-utils: make package private, add code owners, add important notice, change version

### PR DESCRIPTION
**Pull request by flootr**
#### Changes proposed in this Pull Request

* Mark `@automattic/js-utils` as private and change version to `0.1.0`
* Add @Automattic/team-calypso-frameworks as code owners
* Add a note about the purpose of this package and warn about experimental status

Context: p1614937379412800-slack-C7YPUHBB2

#### Testing instructions

* Make sure tests pass
* Do the changes make sense?
* Check for spelling mistakes 😬

### CODE REVIEW

**Comment by sarayourfriend** [comment]
Sorry if I missed it but why are we doing this?

**Comment by flootr** [comment]
> Sorry if I missed it but why are we doing this?

Sorry, I added some context in the PR description 👍🏻

**Comment by scinos** [comment]
FWIW, I think there is value in having this as a separate package, even if we never publish it anywhere.

**Comment by tyxla** [comment]
I don't have strong feelings either way. The only thing I'd suggest is that if we keep it as a package, it shouldn't be a public one, as this isn't ready for public consumption yet.

**Comment by noahtallen** [comment]
Imo, package makes sense vs client/lib, since we may need to use it outside of client.

File: packages/page-template-modal/package.json [hunk 1]
```diff
@@ -28,7 +28,7 @@
 		"src"
 	],
 	"dependencies": {
-		"@automattic/js-utils": "^1.0.0",
+		"@automattic/js-utils": "^0.1.0",
```
Comment (flootr) [inline packages/page-template-modal/package.json]: Note: we should remove the package from here but I'd suggest a follow up /cc @tyxla

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -95,3 +95,6 @@
 /client/state/experiments @Automattic/experimentation-platform
 /client/lib/explat @Automattic/experimentation-platform
 /packages/explat-* @Automattic/experimentation-platform
+
+# Framework
+/packages/js-utils @Automattic/team-calypso-frameworks
```
Comment (tyxla) [inline .github/CODEOWNERS]: Nit: Let's add an EOL so editors don't complain

```suggestion
/packages/js-utils @Automattic/team-calypso-frameworks

```

File: packages/page-template-modal/package.json (hunk 1 above)
Comment (tyxla) [reply packages/page-template-modal/package.json]: Follow-up sounds good 👍  I'll be getting rid of `isNullish` in another PR.

File: packages/js-utils/README.md [hunk 3]
```diff
@@ -1,3 +1,14 @@
...
+## Important note
+
+This package was created to help with Lodash removal in Calypso in the first place and is currently
+not intended to be used outside of Calypso. Functions currently in here may be removed in the future
+without further notice.
+
+Second, we want to think very carefully about what functions to add and which not. A lot of Lodash
+functionality these days can be replaced by native JavaScript code and that should be the premise.
+In case you still want to add a function here please open a Pull Request which describes your use-case
+and ping @Automattic/team-calypso-framework for a review.
```
Comment (tyxla) [inline packages/js-utils/README.md]:
```suggestion
and ping @Automattic/team-calypso-frameworks for a review.
```

**Review by tyxla — APPROVED** [review]
Aside from a couple of nits, LGTM 👍  Thanks!

**Review by noahtallen — APPROVED** [review]
Nice! 🚀

**Pull request merged**